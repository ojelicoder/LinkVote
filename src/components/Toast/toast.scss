@import '../../styles/variables';

.toast{
    position: fixed;
    top: -365px;
    transition: 0.5s ease-in-out;
    width: auto;
    border-radius: 10px;
    -webkit-box-shadow: 0px 0px 15px 1px rgb(0 0 0 / 34%);
    box-shadow: 0px 0px 15px 1px rgb(0 0 0 / 34%);

    &.in{
        transform: translateY(400px);
    }

    &-success{
        background-color: $color-misty-lawn;
        color: $color-mid-green;
        text-align: center;
        padding: 10px 20px;
        font-size: 30px;
        -webkit-box-shadow: inset 0px 0px 39px 8px $color-bright-mint;
        box-shadow: inset 0px 0px 39px 8px $color-light-pastel-green;
        border-radius: 5px;
        border: 1px solid $color-mid-green;
    }
    &-error{
        background-color: $color-faded-red;
        color: $color-white;
        text-align: center;
        padding: 10px 20px;
        font-size: 30px;
        -webkit-box-shadow: inset 0px 0px 39px 8px $color-faded-red;
        box-shadow: inset 0px 0px 39px 8px $color-faded-red;
        border-radius: 5px;
        border: 1px solid $color-faded-red;
    }
    &_content{
        &-up{
            text-transform: uppercase;
            font-weight: 700;
        }
    }
}