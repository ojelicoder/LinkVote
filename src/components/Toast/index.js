import React, { useEffect, useState } from 'react';
import './toast.scss';

const Toast = (props) => {
    const [toastShow, setToastShow] = useState(true),
    {message, type, name, toastClosed} = props;

    useEffect(() => {
        setToastShow(false)

        setTimeout(() => {
            setToastShow(false)
            toastClosed()
        }, 2000);
    }, [toastClosed])

    return (
        <div className={toastShow ? 'toast' : 'toast in'}>
            <div className={type === 'success' ? 'toast-success' : 'toast-error'}>
                <div className='toast_content'>
                    <span className='toast_content-up'>{name}</span> {message}
                </div>
            </div>
        </div>
    )
}

export default Toast;