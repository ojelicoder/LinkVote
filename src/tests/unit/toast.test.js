import React from 'react';
import reactDom from 'react-dom';
import Toast from '../../components/Toast';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const toastProp = {
    message : 'Success process',
    type: 'success',
    name: 'Test link name',
    toastClosed: true
},

wrapper = shallow(<Toast props={toastProp} />);

describe('Toast', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        reactDom.render(<Toast props={toastProp} />, div);
    })

    it('renders DOM elements correctly', () => {
        const components = wrapper.find('div.toast');
        expect(components.length).toBe(1);
    })

    it('matches snapshot', () => {
        const tree = renderer.create(<Toast toastProp={{type: 'Success matches snapshot'}} />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('renders error toast correctly', () => {
        renderer.create(<Toast toastProp={{type: 'error'}} />).toJSON();
        const components = wrapper.find('div.toast-error');
        expect(components.length).toBe(1);
    })
})