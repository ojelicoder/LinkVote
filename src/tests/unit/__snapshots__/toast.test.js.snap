// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast matches snapshot 1`] = `
<div
  className="toast"
>
  <div
    className="toast-error"
  >
    <div
      className="toast_content"
    >
      <span
        className="toast_content-up"
      />
       
    </div>
  </div>
</div>
`;
